packages/components/src/components/input/input.ts

```diff
@@ -388,61 +403,129 @@ export default class SdInput extends SolidElement implements SolidFormControl {
+
+    // Conditional Styles
+    const textSize = this.size === 'sm' ? 'text-sm' : 'text-base';
+    const textColor = {
+      disabled: 'text-neutral-500',
+      readonly: 'text-black',
+      activeInvalid: 'text-error',
+      activeValid: 'text-success',
+      active: 'text-black',
+      invalid: 'text-error',
+      valid: 'text-success',
+      default: 'text-black'
+    }[inputState];
+
+    const borderColor = {
+      disabled: 'border-neutral-500',
+      readonly: 'border-neutral-800',
+      activeInvalid: 'border-error border-2',
+      activeValid: 'border-success border-2',
+      active: 'border-primary border-2',
+      invalid: 'border-error',
+      valid: 'border-success',
+      default: 'border-neutral-800'
+    }[inputState];
+
+    const iconMarginLeft = this.size === 'sm' ? 'ml-1' : 'ml-2';
+    const iconSize = {
+      sm: 'text-base',
+      md: 'text-lg',
+      lg: 'text-xl'
+    }[this.size];
+
+    // Render
     return html`
       <div
         part="form-control"
-        class=${classMap({
-      'form-control': true,
-      'form-control--small': this.size === 'small',
-      'form-control--medium': this.size === 'medium',
-      'form-control--large': this.size === 'large',
-      'form-control--has-label': hasLabel,
-      'form-control--has-help-text': hasHelpText
-    })}
+        class=${cx('form-control', hasLabel && 'form-control--has-label', this.disabled && 'pointer-events-none')}
       >
         <label
           part="form-control-label"
-          class="form-control__label"
+          class=${cx('form-control-label mb-2', hasLabel ? 'inline-block' : 'hidden', textSize)}
           for="input"
           aria-hidden=${hasLabel ? 'false' : 'true'}
         >
-          <slot name="label">${this.label}</slot>
+          <slot name="label">${this.label}${this.required ? '*' : ''}</slot>
         </label>
 
-        <div part="form-control-input" class="form-control-input">
+        <div part="form-control-input" class="form-control-input relative w-full">
+          <div part="border" class=${cx(
+            'absolute w-full h-full pointer-events-none border rounded-[4px]',
+            borderColor
+          )}></div>
           <div
             part="base"
-            class=${classMap({
-      input: true,
-
-      // Sizes
-      'input--small': this.size === 'small',
-      'input--medium': this.size === 'medium',
-      'input--large': this.size === 'large',
-
-      // States
-      'input--pill': this.pill,
-      'input--standard': !this.filled,
-      'input--filled': this.filled,
-      'input--disabled': this.disabled,
-      'input--focused': this.hasFocus,
-      'input--empty': !this.value,
-      'input--no-spin-buttons': this.noSpinButtons,
-      'input--is-firefox': isFirefox
-    })}
+            class=${cx(
+              'px-4 rounded flex flex-row items-center rounded-[4px]',
+              // Vertical Padding
+              this.size === 'lg' ? 'py-2' : 'py-1',
```

Comment:
Is there a specific reason why you've chosen a completely different query here compared to the other components? Doesn't it work like this:
```
{
  /* sizes */
  sm: 'py-1',
  md: 'py-1',
  lg: 'py-2'
}[this.size]
```

For me this one is more readable.

Reply:
One line vs. 5 lines, 2 conditions vs. 3 conditions.  If you prefer class maps, I will use this as a guideline.